lib/Red/AST/Optimizer/AND.pm6

```diff
@@ -36,52 +39,60 @@ multi method optimization-col1(
 }
 
 #| x > 10 AND x < 1 ==> False
-multi method optimization-col1(
+multi method optimize(
     $left  where Red::AST::Ge|Red::AST::Gt,
-    $right where Red::AST::Le|Red::AST::Lt
+    $right where Red::AST::Le|Red::AST::Lt,
+    1
 ) {
     my $lv = $left.args.first(*.^can: "get-value").get-value;
     my $rv = $right.args.first(*.^can: "get-value").get-value;
     return ast-value False if $lv.defined and $rv.defined and $lv > $rv
 }
 
 #| x < 1 AND x > 10 ==> False
-multi method optimization-col1(
+multi method optimize(
     $left  where Red::AST::Le|Red::AST::Lt,
-    $right where Red::AST::Ge|Red::AST::Gt
+    $right where Red::AST::Ge|Red::AST::Gt,
+    1
 ) {
     my $lv = $left.args.first(*.^can: "get-value").get-value;
     my $rv = $right.args.first(*.^can: "get-value").get-value;
     return ast-value False if $lv.defined and $rv.defined and $lv < $rv
 }
 
-#| x < 1 AND NOT(x >= 1) ==> True
-multi method optimization-col1(
+#| a.b AND NOT(a.b) ==> True
+multi method optimize(
     $left  where Red::Column,
-    $right where Red::AST::Not
+    $right where Red::AST::Not,
```

Comment:
I'd go with a regular parameter declaration for these as as the ```where``` makes me think there is something special going on whereas there isn't.